functions.sh

```diff
@@ -4153,6 +4163,36 @@ set_udev_rules() {
     [ "$IAM" == 'suse' ] && suse_version="$IMG_VERSION"
     [ "$suse_version" == "123" ] && suse_netdev_fix
   fi
+
+  # drive write cache (DWC)
+  dwc_current="$(hdparm -W "${DRIVE1}" 2>/dev/null | grep -Po '(?<=write-caching = )(.*?)(?=$)')"
+  dwc_udev_file="${FOLD}/hdd${UDEVPFAD}/65-drive-write-cache.rules"
+  udev_rule=''
+  skip_dwc=0
+  if [[ -n "${DWC_ENABLE}" && "${DWC_ENABLE}" -ge 1 ]]; then
+    if [[ "${dwc_current}" == "not supported" ]]; then
+      hdparm -W "${DRIVE1}" |& debugoutput
+      debug "DWC is not supported by disk, skipping udev rule generation..."
+      skip_dwc=1
+    else
+      # ACTION=="add|change", KERNEL=="sd[a-z]", ATTR{queue/rotational}=="1", RUN+="/usr/bin/hdparm -B 127 -S 12 /dev/%k"
+      if [[ "${DWC_ENABLE}" -eq 1 ]]; then
+        debug "enabling DWC for SSDs..."
+        udev_rule='ACTION=="add|change", KERNEL=="sd[a-z]", ATTR{queue/rotational}=="0", RUN+="/usr/bin/hdparm -W1 /dev/%k"'
+      elif [[ "${DWC_ENABLE}" -eq 2 ]]; then
+        debug "enabling DWC for all disks..."
+        udev_rule='ACTION=="add|change", KERNEL=="sd[a-z]", RUN+="/usr/bin/hdparm -W1 /dev/%k"'
+      else
+        debug "WARN: disabling DWC due to unknown DWC_ENABLE value: ${DWC_ENABLE}"
+        udev_rule='ACTION=="add|change", KERNEL=="sd[a-z]", RUN+="/usr/bin/hdparm -W0 /dev/%k"'
+      fi
+    fi
+  else
+    debug "disabling DWC due to DWC_ENABLE == 0"
+    udev_rule='ACTION=="add|change", KERNEL=="sd[a-z]", RUN+="/usr/bin/hdparm -W0 /dev/%k"'
```

Comment:
Is the absolute path valid on all operating systems we support?

Reply:
good catch, I will verify that and report back

Reply:
should be properly handled now.